File: src/eqn/parser/exception/EqnException.java
package eqn.parser.exception;

public class EqnException extends Exception {

    Exception optionalNestedException = null;

    public EqnException(String exceptionMessage) {
        super(exceptionMessage);
    }

    public EqnException(String exceptionMessage, Exception equationException) {
        super(exceptionMessage);
        this.optionalNestedException = equationException;
    }


    @Override
    public String toString() {
        StringBuffer stringBuffer = new StringBuffer(super.toString());

        if (optionalNestedException != null) {
            stringBuffer.append(" (");
            stringBuffer.append(optionalNestedException.toString());
            stringBuffer.append(')');

        }
        return new String(stringBuffer);
    }
}

File: src/eqn/parser/exception/EqnException.kt
package eqn.parser.exception

open class EqnException : Exception {
    var optionalNestedException: Exception? = null

    constructor(exceptionMessage: String?) : super(exceptionMessage) {}
    constructor(exceptionMessage: String?, equationException: Exception?) : super(exceptionMessage) {
        optionalNestedException = equationException
    }

    override fun toString(): String {
        val stringBuffer = StringBuffer(super.toString())
        if (optionalNestedException != null) {
            stringBuffer.append(" (")
            stringBuffer.append(optionalNestedException.toString())
            stringBuffer.append(')')
        }
        return String(stringBuffer)
    }
}